streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -502,6 +517,13 @@ public StreamsConfig(final Map<?, ?> props) {
         consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, originals().get(BOOTSTRAP_SERVERS_CONFIG));
         // remove deprecate ZK config
         consumerProps.remove(ZOOKEEPER_CONNECT_CONFIG);
+        if (eosEnabled) {
+            if (clientProvidedProps.containsKey(ConsumerConfig.ISOLATION_LEVEL_CONFIG)) {
```

Comment:
What is the benefit of logging a warning instead of throwing an exception? If the user explicitly specified a config and we override it, it makes sense to throw an exception so that the user tests show the issue (i.e. fail fast). The user can then figure out if the config was a mistake or if they misunderstood some aspect and need an alternative solution.

This is what the other clients (e.g. producer) do.